pkg/ingester/ingester.go

```diff
@@ -2152,6 +2168,14 @@ func (i *Ingester) trackInflightQueryRequest() (func(), error) {
 	}
 
 	i.maxInflightQueryRequests.Track(i.inflightQueryRequests.Inc())
+
+	if i.resourceBasedLimiter != nil {
+		if err := i.resourceBasedLimiter.AcceptNewRequest(); err != nil {
+			level.Warn(i.logger).Log("msg", "failed to accept request", "err", err)
+			return nil, fmt.Errorf("failed to query: %s", limiter.ErrResourceLimitReachedStr)
```

Comment:
Do we want to wrap this with gRPC status code? I wonder what's the status code now if we wrap it this way. Is it a 5xx?

Reply:
It will be returned as 422 to the user. I have [an open PR](https://github.com/prometheus/prometheus/pull/16257) in prometheus to be able to customize the engine error, and once that's merged I can change all grpc-wrapped errors to use the grpc status code.